### ISSUE TITLE
Mean IoU is 0.643 after 26400 iterations and initialized from provided model.

### ISSUE THREAD

**Issue by gaopeng-eugene** [DrSleep/tensorflow-deeplab-resnet#6]

**Comment by gaopeng-eugene** [comment]
root@c8393a415c4e:/opt/data/penggao/deep_segmentation/tensorflow-deeplab-resnet# CUDA_VISIBLE_DEVICES=6 python evaluate.py --restore_from ./snapshots_no_index/model.ckpt-26400
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcurand.so locally
WARNING:tensorflow:From evaluate.py:84 in main.: all_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Please use tf.global_variables instead.
I tensorflow/core/common_runtime/gpu/gpu_device.cc:885] Found device 0 with properties: 
name: Tesla P100-SXM2-16GB
major: 6 minor: 0 memoryClockRate (GHz) 1.4805
pciBusID 0000:89:00.0
Total memory: 15.90GiB
Free memory: 13.57GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:906] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla P100-SXM2-16GB, pci bus id: 0000:89:00.0)
WARNING:tensorflow:From evaluate.py:99 in main.: initialize_all_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Use `tf.global_variables_initializer` instead.
WARNING:tensorflow:From evaluate.py:102 in main.: initialize_local_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Use `tf.local_variables_initializer` instead.
Restored model parameters from ./snapshots_no_index/model.ckpt-26400
step 0 	
step 100 	
step 200 	
step 300 	
step 400 	
step 500 	
step 600 	
step 700 	
step 800 	
step 900 	
step 1000 	
step 1100 	
step 1200 	
step 1300 	
step 1400 	
Mean IoU: 0.643

**Comment by gaopeng-eugene** [comment]
Do you have any suggestion on improve the results. There is a big gap between the paper.

**Comment by gaopeng-eugene** [comment]
After 29600 iterations, mIoU is 0.653

**Comment by DrSleep** [comment]
Is it after running 'fine_tune'?

**Comment by gaopeng-eugene** [comment]
0.653 is from deeplab_init_model (30000)

while 0.789 is from deeplab_pretrained_model (4000)

**Comment by DrSleep** [comment]
I haven't really followed the optimisation methods outlined in the paper (e.g., as I remember they use SGD with a particular schedule for the learning rate vs. Adam in this repository), so first you can try to consult the paper on the subject of the training hyperparameters. I think this should give a performance boost.

**Comment by arslan-chaudhry** [comment]
@gaopeng-eugene did you **train** your model from the initialization weights coming from (**deeplab_resnet_init.ckpt**, given in the instructions)? I tried to use that and also converting the init.caffemodel myself as an initialization model, but in both cases my mIoU is around **53%** using the default hyperparameters in this repo. Even after 50k iterations the mIoU is not improving. 

Did you have to make any changes in the training strategy to get to 65.3%? Also, have you tried tweaking the hyperparameters (learning rate in particular) as per paper description (poly strategy), and did it result in any improvements? I am just wondering, given that **adam** is already adapting the learning rate, using SGD with poly learning rate will not make much of a difference. Anyhow, I'll be pleased to know what you discovered?
